src/flag_gems/runtime/device/device.py

```diff
@@ -0,0 +1,51 @@
+import os
+import subprocess
+
+import torch
+import triton
+
+from .. import backend, error
+
+
+class device_ctx:
+    def __init__(self, vendor_name=None):
+        self.vendor_list = backend.vendors_map.keys()
+        self.device_info = self.get_vendor(vendor_name)
+        self.vendor_name, self.device_name, self.cmd, self.vendor = self.device_info
+
+    def get_vendor(self, vendor_name=None) -> tuple:
```

Comment:
I suggest use typing to add type hints and use type hints that works for older versions of python, for example `typing.Tuple`. Also, if it is possible to be clear, `Tuple` of what?